plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/StandardFacetInstallDelegate.java

```diff
@@ -75,6 +75,7 @@ private AppEngineRuntimeInstallJob(IFacetedProject facetedProject) {
       super(Messages.getString("appengine.install.runtime.to.project", // $NON-NLS$
           facetedProject.getProject().getName()));
       this.facetedProject = facetedProject;
+      setRule(facetedProject.getProject());
```

Comment:
I think there should be a way to apply a scheduling rule to some portion of code of a `Job`, rather than applying it to the entire job? If we grab a lock after `waitUntilJsdtIsFixedFacet()`, at least we can get away with this deadlock.